internal/procinfo/pidtouser.go

```diff
@@ -0,0 +1,51 @@
+package procinfo
+
+import (
+	"errors"
+	"github.com/gpuctl/gpuctl/internal/passwd"
+	"os"
+	"strconv"
+	"syscall"
+)
+
+var (
+	ErrorNoUserFound = errors.New("Could not find user that owns the given process from passwd file")
+)
+
+type UidLookup map[uint32]string
+
+func (lookup UidLookup) Get(pid uint64) (string, error) {
+	var zero string
+
+	// get uid
+	filename := "/proc/" + strconv.FormatUint(pid, 10)
+	statcall, err := os.Stat(filename)
+	if err != nil {
+		return zero, err
+	}
+	stat, okay := statcall.Sys().(*syscall.Stat_t)
+	if okay {
+		return "", err
+	}
```

Comment:
This `if` statement confuses me a little bit. I think that we're casting the file info to something OS-specific, and whether that succeeds is written to `okay`? If so, why do we return an empty name if we did cast successfully?

Secondly, I believe that the returned `err` in this case is always nil, as we've just compared it to nil and not changed it after. I think we instead make a new "Couldn't cast" error and return that

Reply:
Yeah, this is wildly broken.

I think the correct code looks like

```go
if !ok {
    return "", ErrUnsuportedOs
}
```

NIT: Also go uses `ok` (not `okay`) for this sort of variable by convention.  grep for `, ok` in [effective go](https://go.dev/doc/effective_go)